ksql-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -100,6 +100,23 @@
    */
   ExecuteResult execute(ConfiguredStatement<?> statement);
 
+  /**
+   * Execute the supplied statement, updating the meta store and registering any query.
+   *
+   * <p>The statement must be executable. See {@link KsqlEngine#isExecutableStatement}.
+   *
+   * <p>If the statement contains a query, then it will be tracked, but not started.
+   *
+   * <p>The statement is executed on the context of the passed ServiceContext to make sure
+   * the caller has permissions rights on the data sources.
```

Comment:
What the interface does with the service context is an implementation detail, I wouldn't specify that it has to be used to check permissions I would just, at this point, say "using a specific ServiceContext"